Search/ObjectToDocumentConverter.php

```diff
@@ -110,8 +115,16 @@ public function objectToDocument(IndexMetadata $metadata, $object)
             $document->setLocale($locale);
         }
 
+        $this->blockValues = [];
         $this->populateDocument($document, $object, $fieldMapping);
 
+        // Adds the merged data of each content-block (even nested-blocks) to the document.
+        if (0 < \count($this->blockValues)) {
+            $mapping = $this->addMappingOptions();
+            $blockValues = \implode(' ', $this->blockValues);
+            $this->addDocumentField($document, 'contentBlocks', $blockValues, $mapping, Field::TYPE_STRING);
```

Comment:
@thomasduenser we should use a field for that - currently we something like that:

```
blocks0title = "Title 1"
blocks1title = "Title 2"
```

I think in the future we should have:

```
blocks_title = "Title 1,Title 2"
```

And that should be done in the field mapping and not here in the converter.

So i think we should change the behaviour of the `ComplexMetatada` to not populate each value as a own field in the index.

Reply:
@alexander-schranz what do you think about the behaviour change?

Reply:
@wachterjohannes i would do that I would store all into one firld aslo g as thex have the same settings. If we support boost we would create a field oer boost level.

Reply:
@alexander-schranz but i think that is a different issue - we could store all the data of the content in an own field! but that should be a configuration - should this field be an own field or not (could be NOT as default).

But to fix this issue to map block properties (with own field like the current behaviour is) there should be an own field for each `sulu.search.field` which could be used to boost for example. But also for that we would have to combine all the values.

Reply:
@wachterjohannes and that is why I currently would just create one "content" field to store all fields into it instead of create seperate fields.

Reply:
@alexander-schranz as i said before :) should be configureable by field - but currently i would just fix the block issue - everything else should be done i a second PR after digging into it!

Reply:
@wachterjohannes @alexander-schranz Then I'll fix the issues in this PR (stateless service, remove strip_tags)? And the other improvements (configuration per field) will be discussed/changed at another time?